pkg/agent/controller/networkpolicy/priority_test.go

```diff
@@ -337,3 +337,22 @@ func TestRevertUpdates(t *testing.T) {
 		})
 	}
 }
+
+func generatePriorities(start, end int32) []types.Priority {
+	priorities := make([]types.Priority, end-start+1)
+	for i := start; i <= end; i++ {
+		priorities[i-start] = types.Priority{TierPriority: 1, PolicyPriority: 5, RulePriority: i - start}
+	}
+	return priorities
+}
+
+func TestRegisterAllOFPriorities(t *testing.T) {
+	pa := newPriorityAssigner(InitialOFPrioritySingleTierPerTable)
+	maxPriorities := generatePriorities(int32(PriorityBottomCNP), int32(PriorityTopCNP))
+	err := pa.RegisterPriorities(maxPriorities)
+	assert.Equalf(t, nil, err, "Error occurred in registering max number of allowed priorities")
+
+	extraPriority := types.Priority{TierPriority: 1, PolicyPriority: 5, RulePriority: int32(PriorityTopCNP) - int32(PriorityBottomCNP) + 1}
+	_, _, _, err = pa.GetOFPriority(extraPriority)
+	assert.Errorf(t, err, "Error should be raised after max number of priorities are registered")
+}
```

Comment:
with this test, won't we go straight to the `insertionPoint-PriorityBottomCNP <= uint16(i)` case (sequential priority assignment)? is it the best test for your function or should we use a bit fewer priorities?

Reply:
That is actually the case I want to test out. For the 'use the insertionPoint to update priority case', it is no different than registering priorities individually and is already well covered in the other test.